lib/fix_ints.nit

```diff
@@ -0,0 +1,789 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This file is free software, which comes along with NIT.  This software is
+# distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without  even  the implied warranty of  MERCHANTABILITY or  FITNESS FOR A
+# PARTICULAR PURPOSE.  You can modify it is you want,  provided this header
+# is kept unaltered, and a notification of the changes is added.
+# You  are  allowed  to  redistribute it and sell it, alone or is a part of
+# another product.
+
+# Integers of fixed-size with C-like semantics
+#
+# Since Nit vows to make life easier for higher-level programming
+# It lacked a bit in low-level int representation
+#
+# This module introduces several fixed-length integers representations
+#
+# You may use it for special occasions when dealing with close-to-the-metal situations
+#
+# Specifically for:
+#
+# * Cryptography-related operations
+# * Virtual Machine implementation
+# * Emulators
+# * etc.
+#
+# All representations included here are akin to C's stdint defined types
+# When using these, bear in mind that calculations on different types here may overflow
+#
+# Also, when converting from one type to another, unless expected, keep in mind that an overflow
+# may happen, and that by doing so, some data may be lost in translation.
+module fix_ints
+
+`{
+	#include <stdint.h>
+`}
+
+redef class Numeric
+	# Returns `self` as `int8_t`
+	fun to_i8: Int8 is abstract
```

Comment:
you can provide a general implementation with `to_int.to_i8` that will benefit other numeric that can to_int (like float that you forgot)